Assets/MixedRealityToolkit-SDK/Features/Diagnostics/DiagnosticBehavor.cs

```diff
@@ -1,26 +1,24 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using Microsoft.MixedReality.Toolkit.Core.EventDatum.Boundary;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.Diagnostics;
 using UnityEngine;
 
 namespace Assets.MixedRealityToolkit_SDK.Features.Diagnostics
 {
-    public class DiagnosticBehavior : MonoBehaviour
+    public class DiagnosticBehavior : MonoBehaviour, IMixedRealityDiagnosticsHandler
     {
-        // Use this for initialization
-        void Start()
-        {
-
-        }
-
-        // Update is called once per frame
-        void Update()
-        {
-
-        }
-
         public bool ShowCpu { get; set; }
```

Comment:
Do these need to be public still? or set able?

Reply:
Nope, good catch

Reply:
Actually*: Needed to set initial values, since I don't see a way to pass as constructor arguments. Looking through documentation now.

Reply:
Mono's don't have constructors. you can set it inline:

`public bool ShowCpu { get; set; } = false;`

Reply:
I'm talking about from the manager on creation.

```
var component = diagnosticVisualization.GetComponent<DiagnosticBehavior>();
component.ShowCpu = ShowCpu;
component.ShowFps = ShowFps;
component.ShowMemory = ShowMemory;

```


Maybe it would be better to rely on the eventing to pass those values? I believe everywhere the property could be created is followed by a `RaiseDiagnosticsChanged()`

Reply:
Yeah as long as you call `RaiseDiagnosticsChanged` after every place you could create it, you should be okay.